File: iosApp/iosApp/Features/Onboarding/OnboardingPreferences.swift
import Foundation

final class OnboardingPreferences: OnboardingStatusStore {
    private let defaults: UserDefaults
    private let key = "onboarding_completed"

    init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
    }

    func hasCompletedOnboarding() -> Bool {
        defaults.bool(forKey: key)
    }

    func setOnboardingCompleted() {
        defaults.set(true, forKey: key)
    }
}

File: androidApp/src/main/kotlin/com/mohamedfaridelsherbini/moventiq/ui/onboarding/OnboardingPreferences.kt
package com.mohamedfaridelsherbini.moventiq.ui.onboarding

import android.content.Context
import androidx.core.content.edit

private const val PREFS_NAME = "moventiq_app_prefs"
private const val KEY_ONBOARDING_COMPLETED = "onboarding_completed"

class OnboardingPreferences(
    context: Context,
) : OnboardingStatusStore {
    private val prefs = context.applicationContext.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    override fun hasCompletedOnboarding(): Boolean =
        prefs.getBoolean(KEY_ONBOARDING_COMPLETED, false)

    override fun setOnboardingCompleted() {
        prefs.edit { putBoolean(KEY_ONBOARDING_COMPLETED, true) }
    }
}
